CONTRIBUTING.md

```diff
@@ -1,73 +1,143 @@
-export SNOWFLAKE_ROLE=ACCOUNTADMIN
-export SNOWFLAKE_REGION=us-west-2
-export TF_ACC=true
-```
+**⚠️ Important ⚠️** We are in progress of cleaning up the repository structure, so beware of the changes in the packages/directories.
+
+## Running the tests locally
 
-You can also read the config from a `~/.snowflake/config` file, although you will still need to set `TF_ACC` to true.
+Currently, we have three main types of tests:
+- SDK unit tests (in directory `pkg/sdk`, files ending with `_test.go`)
+- SDK integration tests (in directory `pkg/sdk/testint`, files ending with `_integration_test.go`)
+- resource/datasource acceptance tests (in directories `pkg/resources` and `pkg/datasources`, files ending with `_acceptance_test.go`)
 
+Both integration and acceptance tests require the connection to Snowflake (some of the tests require multiple accounts).
+
+The preferred way of running particular tests locally is to create a config file `~/.snowflake/config`, with the following content.
 
-~/.snowflake/config
 ```sh
 [default]
-account='TESTACCOUNT'
-user='TEST_USER'
-password='hunter2'
-role='ACCOUNTADMIN'
+account = "<your account in form of organisation-account_name>"
+user = "<your user>"
+password = "<your password>"
+role = "<your role>"
+host="<host of your account, e.g. organisation-account_name.snowflakecomputing.com>"
 ```
 
-**Note: PRs for new resources will not be accepted without passing acceptance tests.**
+To be able to run all the tests you additionally need two more profiles `[secondary_test_account]` and `[incorrect_test_profile]`:
+
+```sh
+[secondary_test_account]
+account = "<your secondary account in form of organisation-account_name2>"
+user = "<your user on the secondary account>"
+password = "<your password on the secondary account>"
+role = "<your role on the secondary account>"
+host="<host of your account, e.g. organisation-account_name2.snowflakecomputing.com>"
+
+[incorrect_test_profile]
+account = "<your account in form of organisation-account_name>"
+user = "<non-existing user>"
+password = "<bad password>"
+role = "<any role, e.g. ACCOUNTADMIN>"
+host="<host of your account, e.g. organisation-account_name.snowflakecomputing.com>"
+```
+
+We are aware that not everyone has access two multiple accounts, so the majority of tests can be run using just one account. The tests setup however, requires both profiles (`default` and `secondary_test_account`) to be present. You can use the same details for `secondary_test_account` as in the `default` one, if you don't plan to run tests requiring multiple accounts.
+
+To run the tests we have three different commands:
+- `make test` run unit and integration tests
+- `make test-acceptance` run acceptance tests
+- `make test-integration` run integration tests
+
+You can run the particular tests form inside your chosen IDE but remember that you have to set `TF_ACC=1` environment variable to run any acceptance tests (the above commands set it for you).
```

Comment:
also having the `TF_LOG=DEBUG` set up helps with making sure that the right SQLs are being run.

Reply:
Good point, I will add this too.